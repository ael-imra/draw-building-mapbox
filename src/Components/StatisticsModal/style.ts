import { SxProps } from '@mui/material';

export const sx: { [key: string]: SxProps } = {
  statiscticsModal: {
    flexFlow: 'column nowrap',
    alignItems: 'center',
    justifyContent: 'center',
    position: 'absolute',
    top: '50%',
    transform: 'translateY(-50%)',
    right: 20,
    width: '300px',
    height: '90%',
    background: 'white',
    borderRadius: '15px',
    zIndex: 99,
    gap: '15px',
  },
  statiscticsTitle: {
    fontSize: 40,
    fontWeight: 600,
    color: '#333333',
  },
  statisticsItem: {
    flexFlow: 'row nowrap',
    alignSelf: 'start',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: '0 40px',
    '& span': {
      fontWeight: 600,
    },
  },
};
