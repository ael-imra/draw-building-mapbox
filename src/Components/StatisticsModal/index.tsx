import { Grid } from '@mui/material';
import { sx } from './style';

interface StatisticsModalProps {
  landArea: number;
  buildingArea: number;
  buildingFloorArea: number;
  volume: number;
  buildingHeight: number;
}

const StatisticsModal = (props: StatisticsModalProps) => {
  const { landArea, buildingArea, buildingFloorArea, volume, buildingHeight } = props;

  return (
    <Grid container sx={sx.statiscticsModal}>
      <Grid sx={sx.statiscticsTitle}>Statistics</Grid>
      <Grid data-testid="statisticsItem" container sx={sx.statisticsItem}>
        Land Area <span>{landArea}</span>
      </Grid>
      <Grid data-testid="statisticsItem" container sx={sx.statisticsItem}>
        Building Area <span>{buildingArea}</span>
      </Grid>
      <Grid data-testid="statisticsItem" container sx={sx.statisticsItem}>
        Building Floor Area <span>{buildingFloorArea}</span>
      </Grid>
      <Grid data-testid="statisticsItem" container sx={sx.statisticsItem}>
        Volume <span>{volume}</span>
      </Grid>
      <Grid data-testid="statisticsItem" container sx={sx.statisticsItem}>
        Building Height <span>{buildingHeight}</span>
      </Grid>
    </Grid>
  );
};

export default StatisticsModal;
