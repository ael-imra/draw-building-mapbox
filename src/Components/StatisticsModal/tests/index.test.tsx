import { cleanup, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import StatisticsModal from '../';

afterEach(cleanup);
describe('StatisticsModal', () => {
  const props = {
    landArea: 1000,
    buildingArea: 2000,
    buildingFloorArea: 3000,
    volume: 4000,
    buildingHeight: 5000,
  };

  it('should render correctly', () => {
    const { asFragment } = render(<StatisticsModal {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render the correct values', () => {
    render(<StatisticsModal {...props} />);

    const items = screen.queryAllByTestId('statisticsItem');

    expect(items.length).toEqual(5);

    expect(items[0]).toHaveTextContent('Land Area 1000');
    expect(items[1]).toHaveTextContent('Building Area 2000');
    expect(items[2]).toHaveTextContent('Building Floor Area 3000');
    expect(items[3]).toHaveTextContent('Volume 4000');
    expect(items[4]).toHaveTextContent('Building Height 5000');
  });
});
