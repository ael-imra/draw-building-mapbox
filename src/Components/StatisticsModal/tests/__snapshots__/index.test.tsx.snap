// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticsModal should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container css-1mt2vs5-MuiGrid-root"
  >
    <div
      class="MuiGrid-root css-18cuka8-MuiGrid-root"
    >
      Statistics
    </div>
    <div
      class="MuiGrid-root MuiGrid-container css-39fura-MuiGrid-root"
      data-testid="statisticsItem"
    >
      Land Area 
      <span>
        1000
      </span>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container css-39fura-MuiGrid-root"
      data-testid="statisticsItem"
    >
      Building Area 
      <span>
        2000
      </span>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container css-39fura-MuiGrid-root"
      data-testid="statisticsItem"
    >
      Building Floor Area 
      <span>
        3000
      </span>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container css-39fura-MuiGrid-root"
      data-testid="statisticsItem"
    >
      Volume 
      <span>
        4000
      </span>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container css-39fura-MuiGrid-root"
      data-testid="statisticsItem"
    >
      Building Height 
      <span>
        5000
      </span>
    </div>
  </div>
</DocumentFragment>
`;
